linkedList.js

```diff
@@ -0,0 +1,372 @@
+
+    return false;
+  }
+
+  findMax() {
+    var currentNode = this.head;
+    var max = this.head.getData();
+    if(!currentNode) {
+      return null;
+    }
+
+    for(var i = 0; i < this.length(); i++) {
+      if(currentNode.getData() > max) {
+        max = currentNode.getData();
+      }
+      currentNode = currentNode.nextNode;
+    }
+
+    return max;
+  }
+
+  findMin() {
+    var currentNode = this.head;
+    var min = this.head.getData();
+
+    if(!currentNode) {
+      return null;
+    }
+
+    for(var i = 0; i < this.length(); i++) {
+      if(currentNode.getData() < min) {
+        min = currentNode.getData();
+      }
+      currentNode = currentNode.nextNode;
+    }
+
+    return min;
+  }
+
+  findNthFromBeginning(index) {
+    var currentNode = this.head;
+    if(!currentNode) {
+      return null;
+    }
+    for(var i = 0; i < index; i++) {
+      currentNode = currentNode.nextNode;
+    }
+
+    return (currentNode ? currentNode.getData() : null);
+  }
+
+  visit() {
+    var printString = "";
+    var currentNode = this.head;
+    if(!currentNode) {
+      return null;
+    }
+
+    for(var i = 0; i < this.length(); i++) {
+      printString += currentNode.getData() + " ";
+      currentNode = currentNode.nextNode;
+    }
+
+    console.log(printString);
+  }
+
+  insertAscending(value) {
+    var currentNode = this.head;
+    var node = new Node(value);
+    var inserted = false;
+
+
+
+    if(!currentNode) {
+      return null;
+    }
+
+    while(currentNode.nextNode) {
+      var nextNode = currentNode.nextNode;
+
+      if(currentNode.getData() < value && nextNode.getData() > value) {
+        var tempNode = currentNode.nextNode;
+        node.nextNode = tempNode;
+        currentNode.nextNode = node;
+        inserted = true;
+
+      }
+      currentNode = currentNode.nextNode;
+    }
+
+    if(!inserted) {
+      currentNode.nextNode = node;
+    }
+  }
+
+  deleteNode(value) {
+    var currentNode = this.head;
+    var nextNode;
+
+    if(!currentNode) {
+      return null;
+    }
+
+    if(this.head.getData() === value ) {
+      nextNode = this.head.nextNode;
+      this.head = nextNode;
+    } else {
+      while(currentNode.nextNode) {
+        nextNode = currentNode.nextNode;
+        if(nextNode.getData() === value ) {
+          var tempNode = nextNode.nextNode;
+          currentNode.nextNode = nextNode.nextNode;
+        }
+        currentNode = currentNode.nextNode;
+      }
+    }
+  }
+
+  reverse() {
+    var currentNode = this.head;
+    var previousNode = null;
+    var tempNode = null;
+
+    if(!currentNode) { return null; }
+
+    while(currentNode) {
+      tempNode = currentNode.nextNode;
+      currentNode.nextNode = previousNode;
+      previousNode = currentNode;
+      currentNode = tempNode;
+    }
+
+    this.head = previousNode;
+  }
+
+  findNthFromEnd(index) {
+    var currentNode = this.head;
+
+    for(var i = 0; i < index; i++) {
+      currentNode = currentNode.nextNode;
```

Comment:
check for null before updating currentNode to currentNode.nextNode. Consider how your code should for empty linked list and for linked list with less than index+1 number of nodes.